### ISSUE TITLE
no way to cleanly express recursive parameters

### ISSUE THREAD

**Issue by seebs** [rails/strong_parameters#251]
Consider an object which belongs_to :parent and has_many :children, and which wishes to accept nested parameters for its children.

A few minutes of searching has turned up a lot of "here's how to unconditionally whitelist the entire tree rooted at a given point", but of course, that's exactly not what we want; we want to actually have working parameter protection which recognizes that, whenever a nested attribute of this type is found, we accept a specific fixed set of attributes for it. Even if one of those is, say, an array of more objects of that type.

I found one example someone wrote called permit_recursive_params, but it seems like this is a use case particularly well-suited to being solved by program logic rather than building huge data structures.

**Comment by rafaelfranca** [comment]
Please use the mailing list or StackOverflow for questions/help, where a wider community will be able to help you. We reserve the issues tracker for issues only.

**Comment by seebs** [comment]
I am confused.

I did not ask a question, I did not request help. I already found a workaround. But so far as I can tell, every observable discussion of this concludes that the right thing to do is something deeply insane, like "just unconditionally whitelist the entire parameter tree starting at the nested parameter", and that is an issue. Maybe it's a documentation issue. Maybe it's a design choice not to support that.

But this certainly looks to me like what I'd usually consider an "issue": A common use case where the public discussion of it consistently involves people recommending deeply insane ideas to work around a thing that's otherwise impossible to achieve.
